src/operator/elementwise_binary_op-inl.h

```diff
@@ -249,13 +294,15 @@ class ElementWiseBinaryOpProp : public OperatorProperty {
     const std::vector<int> &out_data,
     const std::vector<void*> &in_grad) const override {
     switch (GetOpType<ForwardOp>()) {
-      case elembinary::kPlus:
-      case elembinary::kMinus:
-        return {};
-      case elembinary::kMul:
-      case elembinary::kDiv:
-      case elembinary::kPower:
-        return {{out_grad[elembinary::kOut], in_grad[elembinary::kLhs]}};
+    case elembinary::kPlus:
```

Comment:
Actually the current design of backward seems strange. Why use template argument for forward op but switch for backward? How about add a template argument BackwardOp? @antinucleon